demo/dev-tray.ts

```diff
@@ -0,0 +1,180 @@
+import { html, css, LitElement, customElement, property } from 'lit-element';
+import { TimelineMoment } from '../src/interfaces';
+
+@customElement('dev-tray')
+export class DevTray extends LitElement {
+  @property({ type: String }) viewMode: 'open' | 'closed' = 'open';
+
+  @property({ type: Object }) momentsDisplayed: TimelineMoment[] = [];
+
+  directorySubmit(e: Event) {
+    e.preventDefault();
+    const form = e.target as any;
+    const directory = form[0].value;
+    if (!directory) {
+      return;
+    }
+
+    this.dispatchEvent(
+      new CustomEvent('directoryChange', {
+        detail: { directory },
+      })
+    );
+  }
+
+  bannerHeightSubmit(e: Event) {
+    e.preventDefault();
+    const form = e.target as any;
```

Comment:
I will try this again, I specifically remember this `any` bc the compiler would not accept `HTMLFormElement`? it will probably work now...

Reply:
I tried changing it and that was fine, but then it complained about the next line because it doesn't know what `form[0]` is. You just have to tell Typescript what it is:
```ts
    const form = e.target as HTMLFormElement;
    const bannerHeight = (form[0] as HTMLInputElement).value;
```

Reply:
updated

Reply:
Another approach I like using for things like this is the `@query()` decorator. Instead of traversing `e.target`, just make an explicit selector:
```
@query('#myFormInput') myFormInput!: HTMLInputElement;

formChanged() {
  const value = this.myFormInput.value;
}
```